### ISSUE TITLE
[HOLD 16078][$1000] [Android mWeb] Expanded composer is smaller than initial composer on full screen with keyboard

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16446]
On Hold for https://github.com/Expensify/App/issues/16078

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Open chat
3. Open Keyboard
4. Enter multiple lines
5. Expand the composer
6. Observe the full composer is smaller than before

## Expected Result:
The expanding composer shouldn’t be smaller than the composer when not expanding

## Actual Result:
The expanding composer is smaller than the composer when not expanding

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native (  only reproduced on phone which have the big keyboard)  
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.88-0
**Reproducible in staging?:** Needs reproduction
**Reproducible in production?:** Needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

- Samsung A30, Android 10
- Samsung A53, Android 13
- pixel xl

https://user-images.githubusercontent.com/43996225/227223020-5bcf04fb-f1a0-43f9-bb91-cdda9c78b519.mp4

https://user-images.githubusercontent.com/43996225/227223023-b3d986ed-65c5-4e32-959d-630f61d24560.mp4

https://user-images.githubusercontent.com/43996225/227223027-984061da-76b9-421b-a854-49e8bcba64fa.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @tienifr 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1679330011130849?thread_ts=1679302259.813049&cid=C049HHMV9SM

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f2b3ecb55fe13e7b</li>
        <li>Upwork Job ID: 1643602791840743424</li>
        <li>Last Price Increase: 2023-04-05</li>
    </ul>
</details>

**Comment by conorpendergrast** [comment]
Using this to test on Android - https://stackoverflowteams.com/c/expensify/questions/14817

**Comment by conorpendergrast** [comment]
Couldn't reproduce via BrowserStack, will try and get some help to reproduce tomorrow

**Comment by conorpendergrast** [comment]
[Seeking assistance here](https://expensify.slack.com/archives/C01SKUP7QR0/p1679676194363899)

**Comment by strepanier03** [comment]
I tested this for Conor and I cannot recreate.

**Testing notes**

Phone: Pixel 6
OS: Android
Keyboard: Standard Android keyboard

**Screenshots**

![Screenshot_20230324-110520](https://user-images.githubusercontent.com/10925636/227605989-bf9f4b1e-4b6a-40c0-a8cd-84eaaf38de13.png)

----

![Screenshot_20230324-110557](https://user-images.githubusercontent.com/10925636/227606313-05110f1e-cf35-4ab8-b4e7-188e4fb6a5a3.png)

**Comment by conorpendergrast** [comment]
Thanks so much Sheena! 

@tienifr Do you recall any more specific reproduction details that might help us out here?

**Comment by tienifr** [comment]
If you're using GBoard, try to make the keyboard as **tall** as possible.

1. Click on GBoard Settings icon > Preferences
2. Turn on Number row
3. Click Keyboard height
4. Select Extra-tall

Try to use a small screen device. Mine is Pixel XL with 5.5".

I think that this only happens when the expanded composer may overlap the Report Page underneath.

**Comment by MelvinBot** [comment]
@conorpendergrast Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by conorpendergrast** [comment]
I've tried reproducing on a Pixel 3 XL via BrowserStack, with the GBoard, with number row enabled and keyboard height set to extra tall, and the expanded composer is still larger. 

@strepanier03 Could you try once more on a physical device again?

**Comment by strepanier03** [comment]
All I can test on is a Google Pixel 6 pro which has a 6.7 inch screen, I don't have a smaller option. 

I am already using Gboard (it's the default for my phone) and after setting the keyboard to extra tall I am still unable to recreate the behavior. The expanded composer is still larger than the composer not expanded. 

![Screenshot_20230329-171606](https://user-images.githubusercontent.com/10925636/228697607-54c1bb7c-334a-4856-a43b-0ba9c2b7bf5d.png)

**Comment by tienifr** [comment]
@strepanier03 Could you help try again on mWeb on Google Pixel 6 pro? If you still can't reproduce it, we should close this GH issue

**Comment by strepanier03** [comment]
Sure thing @tienifr I'll give that a go now.

**Comment by strepanier03** [comment]
So oddly, on mWeb I can recreate this. The expanded compose field is just the barest tiniest bit smaller than unexpanded. It's almost not noticeable. Below is another side-by-side. 

![Screenshot_20230330-202950](https://user-images.githubusercontent.com/10925636/229016086-da938715-4421-4ed0-98a1-39c959cdb40f.png)

**Comment by conorpendergrast** [comment]
Yeah, I can do that, thanks for reproducing it Sheena! Unassigning you

**Comment by conorpendergrast** [comment]
Actually, I do think we should fix this - it's a reproducible bug (albeit one that probably almost no one will encounter). Let's do ti!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01f2b3ecb55fe13e7b

**Comment by MelvinBot** [comment]
Current assignee @conorpendergrast is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @thesahindia (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MonilBhavsar (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by mountiny** [comment]
Callstack will make a proposal for this one!

**Comment by narefyev91** [comment]
Hi I'm Nicolay from Callstack - expert contributor group - will start investigation in this area

**Comment by MelvinBot** [comment]
@conorpendergrast @koko57 @MonilBhavsar @thesahindia this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
📣 @narefyev91 You have been assigned to this job by @conorpendergrast!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by conorpendergrast** [comment]
Great, assigned you @narefyev91! @mountiny Do you want to be assigned here too? Does anyone else need to be unassigned?

**Comment by mountiny** [comment]
I think @MonilBhavsar can handle this as CME 🙇

**Comment by conorpendergrast** [comment]
Fantastic, and I'll stay from the BugZero side of it

**Comment by narefyev91** [comment]
My updates - detected the issue - found solution to fix it, will provide proposal after internal review

**Comment by MonilBhavsar** [comment]
I agree we should fix this 👍 
Updated the title to explain the problem in a better way.
Thanks @narefyev91!

**Comment by narefyev91** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
The expanding composer is smaller than the composer when not expanding

### What is the root cause of that problem?
On small devices / while using obligate numbers on the top of virtual keyboard / using tall, middle tall and extra tall keyboard  available space between message container and composer applies incorrectly.

### What changes do you think we should make in order to solve the problem?
Page itself is constructed like this when composer is not expanded
```
<div flex = 1> 
   <div1 flex = 1>Messages</div1> 
   <div2>Composer</div2>
</div>
```
When keyboard is large (tested on Pixel device with large keyboard and numbers on top) - `first child div has available height = 129 and composer = 157 (on real device even less)`. When we click to expand composer - we adding` flex: 1` to it. **In that case height should be the same for children - and will be 143 - btw we expect that composer should be bigger (expanded) but in reality it has 143 < 157.**
In any case using flex 1 - will make children 50% vs 50% - we can give composer a little bit more rights. Adding `flexBasis: '25%', ` will give a bit more space for it. And in any other possible cases - like smaller devices, other types of keyboard and etc, expanded composer will be a bit bigger following the standard rules of html

### What alternative solutions did you explore?

**Comment by mountiny** [comment]
@thesahindia what do you think of the proposal?

**Comment by MonilBhavsar** [comment]
> In that case height should be the same for children - and will be 143 - btw we expect that composer should be bigger (expanded) but in reality it has 143 < 157.


and it will be 129 right? or I think I'm not getting it.

**Comment by narefyev91** [comment]
@MonilBhavsar not really - first child has height = 129, composer has height = 157. When we try to expand composer - we adding flex = 1 to it - which indicates that both first child and composer must have the same available space - which will be (129 + 157) / 2 = 143. Applying 143 for each of the child - will increase height for first child and make composer smaller (initial height was 157 and will be 143)
![photo_2023-04-13 10 21 22 (1)](https://user-images.githubusercontent.com/28352309/231684686-7e7dc516-1516-47d5-8940-008989ad981e.jpeg)

You can see on the image above that first child has by default 120 height, and composer 157. Adding flexBasis - we will say that no matter how much space available during devision (50% vs 50%) - composer should occupy a little bit more space than first child

**Comment by MelvinBot** [comment]
@conorpendergrast @narefyev91 @MonilBhavsar @thesahindia this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MonilBhavsar** [comment]
Thanks for the explanation. Makes sense to me now 🙇

**Comment by MelvinBot** [comment]
@conorpendergrast, @narefyev91, @MonilBhavsar, @thesahindia Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@conorpendergrast, @narefyev91, @MonilBhavsar, @thesahindia Still overdue 6 days?! Let's take care of this!

**Comment by MelvinBot** [comment]
@conorpendergrast, @narefyev91, @MonilBhavsar, @thesahindia 8 days overdue is a lot. Should this be a Weekly issue? If so, feel free to change it!

**Comment by MonilBhavsar** [comment]
Yes, let's update to weekly since we're holding on inline emoji suggestions fix

**Comment by conorpendergrast** [comment]
@MonilBhavsar What's the issue that this is held on?

**Comment by mallenexpensify** [comment]
@MonilBhavsar What's the issue that this is held on?

Asking cuz there's a similar issue [here](https://github.com/Expensify/App/issues/18528#issuecomment-1540571350) which I might also want to put on hold

**Comment by mallenexpensify** [comment]
Thanks @thesahindia , I updated the title to `[HOLD #16078]`

**Comment by conorpendergrast** [comment]
Still held on https://github.com/Expensify/App/issues/16078

**Comment by conorpendergrast** [comment]
#16078 is now [on production](https://github.com/Expensify/App/issues/16078#issuecomment-1563877390), so I'm taking this off hold now too

**Comment by trjExpensify** [comment]
Now that it's off hold, let's drop it back to `daily` as per all unheld issues should be in /App. Looks like @narefyev91 [has a PR](https://github.com/Expensify/App/pull/17396) in review for it as well, right? If so, let's add `reviewing` to reflect that if we're proceeding with it.

**Comment by conorpendergrast** [comment]
Oh yesss thanks Tom. I'll wait to confirm before we add the `Reviewing` label

**Comment by narefyev91** [comment]
> Now that it's off hold, let's drop it back to `daily` as per all unheld issues should be in /App. Looks like @narefyev91 [has a PR](https://github.com/Expensify/App/pull/17396) in review for it as well, right? If so, let's add `reviewing` to reflect that if we're proceeding with it.

Yup PR is already there - if nothing really changed around this issue

**Comment by MonilBhavsar** [comment]
I think we still need to put this on HOLD, as we're updating react-native version for `App` and then we'll have a component to make suggestions work line.

**Comment by conorpendergrast** [comment]
Can do! What's the issue for that update, @MonilBhavsar ?

**Comment by MonilBhavsar** [comment]
It should be the same issue #16446. It will have another PR soon

**Comment by conorpendergrast** [comment]
Still on hold for https://github.com/Expensify/App/issues/16078

**Comment by conorpendergrast** [comment]
And, this is still on hold for https://github.com/Expensify/App/issues/16078!

**Comment by MonilBhavsar** [comment]
Still on hold

**Comment by conorpendergrast** [comment]
Still on hold for https://github.com/Expensify/App/issues/16078

**Comment by conorpendergrast** [comment]
On hold for https://github.com/Expensify/App/issues/16078

**Comment by conorpendergrast** [comment]
Still on hold for https://github.com/Expensify/App/issues/16078

**Comment by conorpendergrast** [comment]
Still on hold, but https://github.com/Expensify/App/issues/16078 is now unblocked!

**Comment by conorpendergrast** [comment]
Still held on https://github.com/Expensify/App/issues/16078

**Comment by conorpendergrast** [comment]
@JmillsExpensify I'm on parental leave; re-assigning! This is on hold for https://github.com/Expensify/App/issues/16078, for re-testing to confirm if it's still an issue

**Comment by MonilBhavsar** [comment]
Still on HOLD

**Comment by MonilBhavsar** [comment]
Bumping this to weekly as it is on HOLD

**Comment by MonilBhavsar** [comment]
Still on HOLD

**Comment by JmillsExpensify** [comment]
Looks like we're still on hold for the linked issue. @conorpendergrast Do you mind taking this back as you come back from leave? I'm trying to get my GH assignments down to a more sustainable level.

**Comment by conorpendergrast** [comment]
Yep, and I'll move this to monthly too!

**Comment by conorpendergrast** [comment]
Also I fell on the External button which removed the label, sorry! I'll re-add that if it's needed after re-testing

**Comment by conorpendergrast** [comment]
Still on hold for https://github.com/Expensify/App/issues/16078

**Comment by MonilBhavsar** [comment]
Still on hold

**Comment by conorpendergrast** [comment]
Still on hold for https://github.com/Expensify/App/issues/16078

**Comment by conorpendergrast** [comment]
@MonilBhavsar I'm looking to see what wave this might be part of (once https://github.com/Expensify/App/issues/16078 is off hold). Do you agree that this should be part of #vip-vsb after that? My thinking is that it's mostly related to chat, so that wave seems to make the most sense. Do you have any other ideas? Thanks!

**Comment by MonilBhavsar** [comment]
#vip-vsb sounds good.
Looks like PR for https://github.com/Expensify/App/issues/16078 is in review

**Comment by MonilBhavsar** [comment]
Still on hold

**Comment by dylanexpensify** [comment]
same sameeeee

**Comment by dylanexpensify** [comment]
Held

**Comment by dylanexpensify** [comment]
@MonilBhavsar still on hold?

**Comment by MonilBhavsar** [comment]
Yes! Latest update was here https://github.com/Expensify/App/issues/16078#issuecomment-1963578845

**Comment by dylanexpensify** [comment]
cool cool, ty!!

**Comment by dylanexpensify** [comment]
held

**Comment by dylanexpensify** [comment]
Held

**Comment by dylanexpensify** [comment]
sameeee

**Comment by dylanexpensify** [comment]
hold

**Comment by dylanexpensify** [comment]
Same

**Comment by dylanexpensify** [comment]
Same

**Comment by dylanexpensify** [comment]
Same unfortunately, though not a high priority

**Comment by dylanexpensify** [comment]
held

**Comment by mallenexpensify** [comment]
As discussed in #contributor-plus [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1744830177250489)

Contributor+: @thesahindia due $500 via NewDot.

**Comment by garrettmknight** [comment]
$500 approved for @thesahindia
